core/Archive/ArchiveInvalidator.php

```diff
@@ -192,7 +192,18 @@ public function forgetRememberedArchivedReportsToInvalidate($idSite, Date $date)
 
         // The process pid is added to the end of the entry in order to support multiple concurrent transactions.
         //  So this must be a deleteLike call to get all the entries, where there used to only be one.
-        Option::deleteLike($id . '%');
+        $this->deleteOptionLike($id);
+        Cache::clearCacheGeneral();
+    }
+
+    private function deleteOptionLike($id)
+    {
+        // we're not using deleteLike since it maybe could cause deadlocks see https://github.com/matomo-org/matomo/issues/15545
+        // we want to reduce number of rows scanned and only delete specific primary key
+        $keys = Option::getLike($id . '%');
+        foreach ( $keys as $key => $val) {
+            Option::delete($key);
```

Comment:
Would it make sense to do a `DELETE FROM option WHERE option_name IN (...)` so it's just one query?

Reply:
@diosmosis was thinking about it but then figured it's easier to just reuse existing code for now. Not sure it makes a big difference?

Reply:
I guess it depends on how many invalidated dates there can usually be at one time for a single site.

Reply:
Shouldn't be too many since we regenerate the tracker cache once the flag is written to avoid another request to set this flag? could be a few if there were many requests in parallel maybe. I'll try to have a look but don't really have time.

Reply:
changed it @diosmosis